--- a/src/core/SConscript
+++ b/src/core/SConscript
@@ -17,7 +17,7 @@
 import os
 import os.path
 
-Import("env","libsonic","libhts_engine")
+Import("env","libhts_engine")
 if env["enable_mage"]:
 	Import("libmage")
 local_env=env.Clone()
@@ -66,7 +66,7 @@
 		 "pitch.cpp"]
 if local_env["enable_mage"]:
 	src.append("mage_hts_engine_impl.cpp")
-for lib in [libhts_engine,libsonic]:
+for lib in [libhts_engine]:
 	src.extend(lib)
 if local_env["enable_mage"]:
 	src.extend(libmage)
--- a/src/third-party/SConscript
+++ b/src/third-party/SConscript
@@ -18,7 +18,7 @@
 
 Import("env")
 
-subdirs=["sonic","hts_engine"]
+subdirs=["hts_engine"]
 if env["enable_mage"]:
 	subdirs.append("mage")
 for subdir in subdirs:
--- a/SConstruct
+++ b/SConstruct
@@ -177,7 +177,7 @@
     else:
         env_args["tools"]=["default","installer"]
     env_args["tools"].extend(["textfile","library"])
-    env_args["LIBS"]=[]
+    env_args["LIBS"]=["-lsonic"]
     env_args["package_name"]="RHVoice"
     env_args["CPPDEFINES"]=[("RHVOICE","1")]
     env=Environment(**env_args)
@@ -276,7 +276,7 @@
             conf.CheckSpdVersion()
     else:
         env["audio_libs"]=[]
-#        has_giomm=conf.CheckPKG("giomm-2.4")
+        has_giomm=conf.CheckPKG("giomm-2.4")
     if env["PLATFORM"]=="win32":
         env.AppendUnique(LIBS="kernel32")
     conf.Finish()
